### PR TITLE
Fix bug The helper "dialog" is not defined (Undefined DialogHelper Class) (NEW)

### LINKED ISSUE #4327: The helper "dialog" is not defined (Undefined DialogHelper Class)
#### Sonata packages

```bash
$ composer show sonata-project/*
sonata-project/admin-bundle              3.13.0 The missing Symfony Admin Generator
sonata-project/block-bundle              3.3.0  Symfony SonataBlockBundle
sonata-project/cache                     1.0.7  Cache library
sonata-project/core-bundle               3.2.0  Symfony SonataCoreBundle
sonata-project/doctrine-orm-admin-bundle 3.1.3  Symfony Sonata / Integrate Doctrine ORM into the SonataAdminBundle
sonata-project/exporter                  1.7.1  Lightweight Exporter library
```

#### Symfony packages

```bash
$ composer show symfony/*
symfony/monolog-bundle     v3.0.3 Symfony MonologBundle
symfony/phpunit-bridge     v3.2.3 Symfony PHPUnit Bridge
symfony/polyfill-apcu      v1.3.0 Symfony polyfill backporting apcu_* functions to lower PHP versions
symfony/polyfill-intl-icu  v1.3.0 Symfony polyfill for intl's ICU-related data and classes
symfony/polyfill-mbstring  v1.3.0 Symfony polyfill for the Mbstring extension
symfony/polyfill-php56     v1.3.0 Symfony polyfill backporting some PHP 5.6+ features to lower PHP versions
symfony/polyfill-php70     v1.3.0 Symfony polyfill backporting some PHP 7.0+ features to lower PHP versions
symfony/polyfill-util      v1.3.0 Symfony utilities for portability of PHP codes
symfony/security-acl       v3.0.0 Symfony Security Component - ACL (Access Control List)
symfony/swiftmailer-bundle v2.4.2 Symfony SwiftmailerBundle
symfony/symfony            v3.2.3 The Symfony PHP framework
```

#### PHP version

```bash
$ php -v
PHP 7.1.1-1+deb.sury.org~trusty+1 (cli) (built: Jan 20 2017 09:43:29) ( NTS )
```

## Subject

Undefined DialogHelper Class:

**\Sonata\AdminBundle\Command\GenerateObjectAclCommand::execute**
```
    /**
     * {@inheritdoc}
     */
    public function execute(InputInterface $input, OutputInterface $output)
    {
        /** @var $dialog \Symfony\Component\Console\Helper\DialogHelper */
        $dialog = $this->getHelperSet()->get('dialog');
```

## Steps to reproduce
`$ php bin/console sonata:admin:generate-object-acl `

## Expected results
All happy

## Actual results

```
  [Symfony\Component\Console\Exception\InvalidArgumentException]
  The helper "dialog" is not defined.
```

### PR DESCRIPTION
**Pull request by vedro-compota**
I am targeting this branch, because my PR is a patch.

Closes #4327

## Changelog

```markdown
### Added
- Command/QuestionableCommand.php

### Changed
 - Command/GenerateAdminCommand.php 
 - Command/GenerateObjectAclCommand.php 
 - Command/QuestionableCommand.php 

```

## To do

- [ ] Update the tests
- [ ] Fix codestyle
- [ ] Add an upgrade note

## Subject

It'll fix #4327

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -60,4 +60,4 @@ Resources/public/vendor/jqueryui/themes/vader
 Resources/public/vendor/jquery/src
 
 Resources/public/vendor/jquery.scrollTo/demo
-Resources/public/vendor/jquery.scrollTo/tests
+Resources/public/vendor/jquery.scrollTo/tests
```
Comment (OskarStark) [inline .gitignore]: please revert this

File: Command/QuestionableCommand.php [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+namespace Sonata\AdminBundle\Command;
+
+use Sensio\Bundle\GeneratorBundle\Command\Helper\DialogHelper;
+use Sensio\Bundle\GeneratorBundle\Command\Helper\QuestionHelper;
+use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\ConfirmationQuestion;
+use Symfony\Component\Console\Question\Question;
+
```
Comment (OskarStark) [inline Command/QuestionableCommand.php]: please add @author tag

File: Command/QuestionableCommand.php [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+abstract class QuestionableCommand extends ContainerAwareCommand
```
Comment (OskarStark) [inline Command/QuestionableCommand.php]: please add tests for this command

**Comment by greg0ire** [comment]
> - Added some `Class::newMethod` to do great stuff

WAT

**Comment by OskarStark** [comment]
Please check StyleCI and fullfill the PR header 👍

File: .gitignore [hunk 4]
```diff
@@ -60,4 +60,3 @@ Resources/public/vendor/jqueryui/themes/vader
 Resources/public/vendor/jquery/src
 
 Resources/public/vendor/jquery.scrollTo/demo
-Resources/public/vendor/jquery.scrollTo/tests
```
Comment (greg0ire) [inline .gitignore]: What does this have to do with the problem at hand?

File: Command/QuestionableCommand.php [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    /**
+     * @param InputInterface  $input
+     * @param OutputInterface $output
+     * @param string          $questionText
+     * @param string          $default
+     * @param string          $separator
+     *
+     * @return string
+     */
+    final public function askConfirmation(InputInterface $input, OutputInterface $output, $questionText, $default, $separator)
```
Comment (core23) [inline Command/QuestionableCommand.php]: Can we make this protected?

File: Command/QuestionableCommand.php [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+            $separator
+        ), ($default === 'no' ? false : true));
+
+        return $questionHelper->ask($input, $output, $question);
+    }
+
+    /**
+     * @return QuestionHelper|DialogHelper
+     */
+    final public function getQuestionHelper()
```
Comment (core23) [inline Command/QuestionableCommand.php]: Protected?

**Comment by greg0ire** [comment]
I think we can close this one in favor of #4339 , correct me if I'm wrong.

**Comment by vedro-compota** [comment]
@greg0ire  you're right. Close it -- all changes are done in #4339
